framework/src/base/MooseApp.C

```diff
@@ -562,20 +600,45 @@ MooseApp::runInputFile()
 
   _action_warehouse.executeAllActions();
 
-  if (isParamValid("mesh_only"))
+  if (getParam<std::string>("split_mesh") != "")
```

Comment:
I'd recommend that you break out this new logic into a subroutine.

Reply:
Also, you probably don't want to run all Actions when doing the splits. Building the equation system might be really expensive for instance. A much better design would be to run the subset of Actions needed to build the mesh and then do the splits.

Reply:
I'll piggy back on mesh-only logic.